android-components/components/feature/fxsuggest/src/main/java/mozilla/components/feature/fxsuggest/FxSuggestStorage.kt

```diff
@@ -0,0 +1,68 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.fxsuggest
+
+import android.content.Context
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers.IO
+import kotlinx.coroutines.cancelChildren
+import kotlinx.coroutines.withContext
+import mozilla.appservices.suggest.SuggestIngestionConstraints
+import mozilla.appservices.suggest.SuggestStore
+import mozilla.appservices.suggest.Suggestion
+import mozilla.appservices.suggest.SuggestionQuery
+import mozilla.components.support.base.coroutines.Dispatchers.Cached
+
+/**
+ * A coroutine-aware wrapper around the synchronous [SuggestStore] interface.
+ *
+ * @property context The Android application context.
+ */
+class FxSuggestStorage(
+    context: Context,
+) {
+    // Lazily initializes the store on first use. `Context.getDatabasePath()` does I/O, so this
+    // value must be accessed in the read or write scope.
+    internal val store: Lazy<SuggestStore> = lazy {
+        SuggestStore(context.getDatabasePath(DATABASE_NAME).absolutePath)
+    }
```

Comment:
Using the `by` delegate keyword here should remove the need for all the explicit `.value` unwrapping
```suggestion
    internal val store: SuggestStore by lazy {
        SuggestStore(context.getDatabasePath(DATABASE_NAME).absolutePath)
    }
```

Reply:
Though I see that it might be necessary for `.cancelReads`

Reply:
We chatted about this on Zoom yesterday, but yep, that's exactly why I used `Lazy<SuggestStore>` instead of delegation—I didn't want `cancelReads` to trigger initialization 😄 (In practice, though, I think it'll always be initialized by the time we call `cancelReads`).